src/ImageSharp/Formats/Jpeg/Components/Decoder/ColorConverters/JpegColorConverter.cs

```diff
@@ -190,45 +195,90 @@ internal struct Vector4Octet
             /// </summary>
             public void Pack(ref Vector4Pair r, ref Vector4Pair g, ref Vector4Pair b)
             {
-                this.V0.X = r.A.X;
-                this.V0.Y = g.A.X;
-                this.V0.Z = b.A.X;
-                this.V0.W = 1f;
-
-                this.V1.X = r.A.Y;
-                this.V1.Y = g.A.Y;
-                this.V1.Z = b.A.Y;
-                this.V1.W = 1f;
-
-                this.V2.X = r.A.Z;
-                this.V2.Y = g.A.Z;
-                this.V2.Z = b.A.Z;
-                this.V2.W = 1f;
-
-                this.V3.X = r.A.W;
-                this.V3.Y = g.A.W;
-                this.V3.Z = b.A.W;
-                this.V3.W = 1f;
-
-                this.V4.X = r.B.X;
-                this.V4.Y = g.B.X;
-                this.V4.Z = b.B.X;
-                this.V4.W = 1f;
-
-                this.V5.X = r.B.Y;
-                this.V5.Y = g.B.Y;
-                this.V5.Z = b.B.Y;
-                this.V5.W = 1f;
-
-                this.V6.X = r.B.Z;
-                this.V6.Y = g.B.Z;
-                this.V6.Z = b.B.Z;
-                this.V6.W = 1f;
-
-                this.V7.X = r.B.W;
-                this.V7.Y = g.B.W;
-                this.V7.Z = b.B.W;
-                this.V7.W = 1f;
+#if SUPPORTS_RUNTIME_INTRINSICS
+                if (Avx2.IsSupported)
+                {
+                    Vector4 vo = Vector4.One;
+                    Vector128<float> valpha = Unsafe.As<Vector4, Vector128<float>>(ref vo);
+
+                    ref byte control = ref MemoryMarshal.GetReference(SimdUtils.HwIntrinsics.PermuteMaskDeinterleave8x32);
+                    Vector256<int> vcontrol = Unsafe.As<byte, Vector256<int>>(ref control);
```

Comment:
By having two versions of `Pack`, we can move these loads outside of the for loop calling `Pack`.

(Not necessarily a suggestion for this PR since it extends the scope quite a lot)